app/models/tutorial.rb

```diff
@@ -79,7 +79,19 @@ def teams_to_csv(assignment)
       end
     end
   end
-
+  def submissions_count_for(assignment)
+    Submission.where(tutorial: t, assignment: assignment)
+              .where.not(manuscript_data: nil).size 
+  end
+  def too_late_count_for(assignment)
+    Submission.where(tutorial: self, assignment: assignment)
+              .where.not(manuscript_data: nil)
+              .select { |s| s.too_late? }.size
+  end
+  def corrected_count_for(assignment)
+    Submission.where(tutorial: self, assignment: assignment)
+              .where.not(correction_data: nil).size 
+  end
```

Comment:
This code seems to not in use and be replaced by a portion in `models/submissions.rb`. You can remove it.